### PR TITLE
CLDC-1272: Handles unicode characters in postcode

### PR DESCRIPTION
**Pull request by smeny**

### CODE REVIEW

File: app/models/case_log.rb [hunk 1]
```diff
@@ -718,4 +723,8 @@ def soft_value_for_period(value)
 
     (value * 52 / num_of_weeks).round(2)
   end
+
+  def upcase_and_remove_whitespace(string)
+    string.present? ? string.upcase.gsub(/\s+/, "") : string
```
Comment (moarpheus) [inline app/models/case_log.rb:728]: `string.upcase.gsub(/\s+/, "") || string` should do the trick too

**Review by moarpheus — APPROVED** [review]

File: app/models/case_log.rb (hunk 1 above)
Comment (smeny) [reply app/models/case_log.rb:728]: I like it but this wouldn't protect a call made with a nil value.

**Comment by baarkerlounger** [comment]
Should we not rather swap the order so that Postcodes are validated before that method is called rather than validating twice?

File: spec/models/case_log_spec.rb [hunk 2]
```diff
@@ -1996,5 +1996,13 @@ def previous_postcode_fields
         expect(case_log.plural_gender_for_person_3).to be_nil
       end
     end
+
+    context "when a postcode contains unicode characters" do
+      let(:case_log) { FactoryBot.build(:case_log, postcode_full: "SR81LS\u00A0") }
+
+      it "triggers a validation error" do
+        expect { case_log.save! }.to raise_error(ActiveRecord::RecordInvalid, /Enter a postcode in the correct format/)
```
Comment (baarkerlounger) [inline spec/models/case_log_spec.rb:2004]: Probably want to be referencing a translation string here?

**Comment by baarkerlounger** [comment]
Order is because we need the value set before we can be sure we can validate it

**Pull request merged**